eq-author/src/App/publish/PublishPage.test.js

```diff
@@ -0,0 +1,208 @@
+    };
+    props = {
+      match: {
+        params: {
+          questionnaireId: "12a34567-b8c9-0123-4de5-fg67h8901234",
+        },
+      },
+    };
+  });
+  it("should have the 'Publish' button disabled until all fields are populated", () => {
+    const PublishPage = render(
+      <MeContext.Provider value={{ me: user }}>
+        <UnwrappedPublishPage {...props} />
+      </MeContext.Provider>
+    );
+
+    expect(PublishPage.getByLabelText("Form type").value).toBe("");
+    expect(PublishPage.getByLabelText("Survey ID").value).toBe("");
+    expect(
+      PublishPage.getByTestId("publish-survey-button").disabled
+    ).toBeTruthy();
+  });
+  it("should fire a mutation to save the text within each input field on blur", async () => {
+    queryWasCalled = false;
+    const newProps = {
+      ...props,
+      publishDetails: {
+        questionnaire: {
+          publishDetails: {
+            surveyId: "123",
+            formType: {
+              ONS: "456",
+            },
+          },
+        },
+      },
+    };
+
+    const mocks = [
+      {
+        request: {
+          query: UPDATE_PUBLISH_DETAILS,
+          variables: {
+            input: {
+              questionnaireId: props.match.params.questionnaireId,
+              surveyId: "123",
+              formType: { ONS: "987" },
+            },
+          },
+        },
+        result: () => {
+          queryWasCalled = true;
+          return {
+            data: {
+              updatePublishDetails: {
+                id: "123-456-789",
+                title: "Waterloo Road",
+                publishDetails: {
+                  surveyId: "123",
+                  formType: "987",
+                  __typename: "Object",
+                },
+                __typename: "Object",
+              },
+            },
+          };
+        },
+      },
+    ];
+
+    const PublishPage = render(
+      <MeContext.Provider value={{ me: user }}>
+        <UnwrappedPublishPage {...newProps} />
+      </MeContext.Provider>,
+      { mocks }
+    );
+
+    fireEvent.change(PublishPage.getByLabelText("Form type"), {
+      target: { value: "987" },
+    });
+    fireEvent.blur(PublishPage.getByLabelText("Form type"));
+    await flushPromises();
+    expect(queryWasCalled).toBeTruthy();
+  });
+  it("should populate the fields from values stored within the database on every render", () => {
+    const newProps = {
+      ...props,
+      publishDetails: {
+        questionnaire: {
+          publishDetails: {
+            surveyId: "123",
+            formType: {
+              ONS: "456",
+            },
+          },
+        },
+      },
+    };
+
+    const PublishPage = render(
+      <MeContext.Provider value={{ me: user }}>
+        <UnwrappedPublishPage {...newProps} />
+      </MeContext.Provider>
+    );
+
+    expect(PublishPage.getByLabelText("Form type").value).toBe("456");
+    expect(PublishPage.getByLabelText("Survey ID").value).toBe("123");
+
+    //pass
```

Comment:
In Python, you would write `pass;` to have an empty function, as otherwise it would complain when you try to compile (if you didn't already know). I was using `//pass` as a reference to myself when first writing the tests as JS doesn't have this afaik. 

But, it's just a useless comment that I should have removed. I'll get on that, haha.